src/SolidStateDetector/SolidStateDetector.jl

```diff
@@ -296,88 +297,41 @@ function generate_random_startpositions(d::SolidStateDetector{T}, n::Int, Volume
-                n_grid_points::Int = g_imax - g_imin + 1
-                n::Int = Int(round(Δg / stepsize))
-                if n > 2 * n_grid_points
-                    stepsize = Δg / (4 * n_grid_points)
-                end
-            else
-                stepsize = 1
-            end
-            if iszero(stepsize) stepsize = 1 end
-            push!(stepsizes, stepsize)
-        end
-        append!(samples, filter( x-> x in object.geometry, sample(g, stepsizes)) )
-    end
-    object_gridpoints = unique!([find_closest_gridpoint(sample_point, grid) for sample_point in samples])
-    return object_gridpoints
+function sample(obj::AbstractObject{T}, g::Grid{T,3,Cartesian,AT})::Vector{CartesianPoint{T}} where {T,AT}
+    g_mid::CartesianTicksTuple{T} = ( x = _get_mid_ticks(g[1].ticks), y = _get_mid_ticks(g[2].ticks), z = _get_mid_ticks(g[3].ticks))
+    #filter!(p -> p in obj, 
+    samples::Vector{CartesianPoint{T}} = vcat([sample(surf, g_mid) for surf in obj.decomposed_surfaces]...)
+    unique!(samples)
+    #)
 end
 
-
-function paint_object(det::SolidStateDetector{T}, object::AbstractObject{T}, grid::CylindricalGrid{T}, ::Val{:φ}, φ::T )  where {T <: SSDFloat}
-    closest_φ_idx=searchsortednearest(grid.axes[2].ticks, φ)
-    stepsize::Vector{T}= [minimum(diff(grid.axes[1].ticks)), IntervalSets.width(grid.axes[2].interval) == 0.0 ? 0.05236 : minimum(diff(grid.axes[2].ticks)), minimum(diff(grid.axes[3].ticks))]
-    stepsize /= 2
-    samples = filter(x-> x in object.geometry, vcat([sample(g, stepsize) for g in object.geometry_positive]...))
-    object_gridpoints = unique!([find_closest_gridpoint(sample_point,grid) for sample_point in samples])
-    return filter(x -> x[2]==closest_φ_idx, object_gridpoints)
-end
-function paint_object(det::SolidStateDetector{T}, object::AbstractObject{T}, grid::CylindricalGrid{T}, ::Val{:r}, r::T )  where {T <: SSDFloat}
-    return CartesianPoint{T}[]
-end
-function paint_object(det::SolidStateDetector{T}, object::AbstractObject{T}, grid::Grid{T}, ::Val{:z}, z::T )  where {T <: SSDFloat}
-    return CartesianPoint{T}[]
+function sample(obj::AbstractObject{T}, g::Grid{T,3,Cylindrical,AT})::Vector{CylindricalPoint{T}} where {T,AT}
+    g_mid::CylindricalTicksTuple{T} = ( r = _get_mid_ticks(g[1].ticks), φ = _get_mid_ticks(g[2].ticks), z = _get_mid_ticks(g[3].ticks))
+    #filter!(p -> p in obj, 
+    samples::Vector{CylindricalPoint{T}} = vcat([sample(surf, g_mid) for surf in obj.decomposed_surfaces]...)
+    unique!(samples)
+    #)
 end
-function paint_object(det::SolidStateDetector{T}, object::AbstractObject{T}, grid::CartesianGrid{T}, ::Val{:x}, x::T )  where {T <: SSDFloat}
-    return CartesianPoint{T}[]
+
+# TODO: if mid ticks are needed: uncomment. If not: remove
+@inline function _get_mid_ticks(gr::Vector{T})::Vector{T} where {T}
+    gr
+    # gr_mid::Vector{T} = Vector{T}(undef, 2 * length(gr) - 1)
+    # gr_mid[1:2:end] = gr
+    # gr_mid[2:2:end] = midpoints(gr)
+    # gr_mid
 end
```

Comment:
This function is used to add more ticks to sample.
The commented lines would allow sampling also the midpoints of the grid ticks.